### ISSUE TITLE
Finish the AnonCreds Api interface

### ISSUE THREAD

**Issue by TimoGlastra** [hyperledger/aries-framework-javascript#1128]
1. [x] Finish https://github.com/animo/anoncreds-interfaces/blob/main/src/AnonCredsApi.ts
2. [x] Add TODO methods
3. [ ] ~~Add methods related to revocation (revokeCredential mostly)~~ (postponed for now)

**Comment by TimoGlastra** [comment]
Here's the latest version of the anoncreds api (will keep them updated here). I've just focused on the functionality as is currently available in AFJ. This means we don't support creation of revocation registries etc.. This seems like a lower priority task.

AnonCreds API

```ts
import {
  CreateAnonCredsCredentialDefinitionOptions,
  CreateAnonCredsCredentialDefinitionReturn,
  CreateAnonCredsSchemaOptions,
  CreateAnonCredsSchemaReturn,
} from "./AnonCredsApiOptions";
import { AnonCredsCredentialDefinition, AnonCredsSchema } from "./models/vdr";

export interface AnonCredsApi {
  /**
   * Retrieve an AnonCreds schema from the registry associated with the `schemaId`.
   */
  getSchema(schemaId: string): Promise<AnonCredsSchema>;

  /**
   * Create a schema with the given `name`, `version` and `attrNames`, and register it
   * on the registry associated with the `issuerId`.
   */
  createSchema(
    options: CreateAnonCredsSchemaOptions
  ): Promise<CreateAnonCredsSchemaReturn>;

  /**
   * Retrieve an AnonCreds credential definition from the registry associated with the
   * `credentialDefinitionId`.
   */
  getCredentialDefinition(
    credentialDefinitionId: string
  ): Promise<AnonCredsCredentialDefinition>;

  /**
   * Create a credential definition for the schema with the given `schemaId`,
   * and register it on the registry associated with the `issuerId`.
   */
  createCredentialDefinition(
    options: CreateAnonCredsCredentialDefinitionOptions
  ): Promise<CreateAnonCredsCredentialDefinitionReturn>;
}
```

AnonCredsApiOptions:

```ts
import { AnonCredsCredentialDefinition, AnonCredsSchema } from "./models/vdr";

export interface CreateAnonCredsSchemaOptions {
  issuerId: string;
  schema: AnonCredsSchema;
}

export interface CreateAnonCredsSchemaReturn {
  schemaId: string;
  schema: AnonCredsSchema;
}

export interface CreateAnonCredsCredentialDefinitionOptions {
  issuerId: string;
  credentialDefinition: {
    schemaId: string;
    tag: string;

    // NOTE: for now we will support the same feature set as is present in AFJ. This
    // means that issuance of revocable credentials is not supported. Once we add support
    // for issuance of revocable credentials we can add it to the options here.
    // If `revocationRegistrySize` is set and a positive number, a revocation registry will also
    // be created and registered on the registry. A tails file will be generated for the revocation
    // registry, and uploaded somewhere. Whenever the revocation registry is full, we will need automatically
    // create a new revocation registry. (TODO: how does this work with endorsement?)
    // revocationRegistrySize?: number;
  };
}

export interface CreateAnonCredsCredentialDefinitionReturn {
  credentialDefinitionId: string;
  credentialDefinition: AnonCredsCredentialDefinition;
}
```

**Comment by TimoGlastra** [comment]
@karimStekelenburg once #1118 is merged I'll add this in a follow up PR
